apis/v1alpha1/validation.go

```diff
@@ -97,8 +97,18 @@ func ValidateCLIRequest(r *CLIRequest) (retErr error) {
 func checkCommand(c string) error {
 	for _, ch := range c {
 		if _, ok := invalidCommandOperators[ch]; ok {
-			return fmt.Errorf("invalid command operator %q", ch)
+			return fmt.Errorf("contains invalid command operators in %q", getMapKeys(invalidCommandOperators))
 		}
 	}
 	return nil
 }
+
+func getMapKeys(m map[rune]struct{}) []rune {
+	keys := make([]rune, len(m))
+	i := 0
+	for k := range invalidCommandOperators {
+		keys[i] = k
+		i++
+	}
+	return keys
+}
```

Comment:
It's more idiomatic to do:

```suggestion
func getMapKeys(m map[rune]struct{}) []rune {
	keys := make([]rune, 0, len(m))
	for k := range invalidCommandOperators {
		keys = append(keys, k)
	}
	return keys
}
```

Reply:
However, that being said, we probably don't need this. A better error message would be "disallowed character '&' at position 12:5", which we can get while iterating the string.

Reply:
Xiyue and I discussed this and think that it might be better to show all the invalid operators in the command, so that people can resolve all the errors once and for all. I suggested this way so that we can stop early and provide more hint about what chars are not allowed. What do you think?

Reply:
```go
func checkCommand(c string) error {
	var merr error
	scanner := bufio.NewScanner(strings.NewReader(c))
	for row := 1; scanner.Scan(); row++ {
		line := scanner.Text()
		for col, r := range line {
			if _, ok := invalidCommandOperators[r]; ok {
				merr = errors.Join(merr, fmt.Errorf("disallowed command character %q at %d:%d", r, row, col))
			}
		}
	}
	return merr
}
```

That'll give output like:

```text
disallowed command character '&' at 3:18
disallowed command character '&' at 3:19
disallowed command character '|' at 3:29
disallowed command character '|' at 3:30
```

Reply:
Yeah, I want to stop early though, the cli command is short, the invalidCommandOperators is also short, just a list telling user what is not allowed is enough, but again they are short so not much different performance wise.  Do we really need Scanner?

Reply:
I don't think we can guarantee the command is going to be short. The scanner is for getting line number. Even if I have a really short command, it's going to be really annoying to find that character in a multi-line command.

Reply:
Cool, updating!